Build: Move `lib/csv` to packages

ryelle opened the pull request:
Move the `lib/csv` utilities to a lerna package. This will now be `@woocommerce/csv`. There should be no functionality change.

To test:

- Build the project, make sure it succeeds.
- Check that downloading report tables still works.

ryelle committed:
Move CSV to packages

ryelle committed:
Update CSV imports

ryelle committed:
Add some basic usage documentation

ryelle committed:
Add csv lib to the enqueued assets

Aljullu commented on packages/csv/README.md:
```diff
@@ -0,0 +1,29 @@
+	// Triggers a browser UI to save a file, named the first argument, with the contents of
+	// the second argument.
+	downloadCSVFile(
+		// Create a file name based on the report title and optional query. Will return a
```
`downloadCSVFile()` can be used for any kind of CSV data, because it just requires the file name and the CSV contents already formatted. Same for `generateCSVFileName()`.

`generateCSVDataFromTable()` requires a `headers` and `rows` params (headers must have a `label` attribute and rows must have a `value` attribute), but as long as this is respected, any tabular data --not only reports-- could use this function.

ryelle committed:
Change name to `csv-export`

ryelle committed:
Update webpack build process to allow for package names with dashes

ryelle commented:
> would csv-export be a more appropriate name?

Yeah, that's a better name. I've updated that in https://github.com/woocommerce/wc-admin/pull/756/commits/8d62d4147b9a44284adf3ebb6f305d1ed25a04be, and then needed to add support for packages with dashes in the name (https://github.com/woocommerce/wc-admin/pull/756/commits/9d10cdb421fc38736455e4cbe009f7ab232178cf). The package name and `wc.__` no longer matched, so now we convert dashed names to `camelCase` on the `wc` global. `wc.csv` is now `wc.csvExport`, though the package is still included via `@woocommerce/csv-export`.

I've also updated the documentation slightly to make it clear that these functions can be used independently, and for things other than reports.

ryelle committed:
Fix capitalization

ryelle committed:
Update readme with docs extracted from jsdoc

ryelle committed:
Fix merge conflict commit 😣

ryelle commented:
@belcherj If you're still looking to review something, this is ready 😁

ryelle committed:
Check that generateCSVFileName is called with correct params

timmyc reviewed (approved):
This is testing out well for me, and now I can add csv export to the WCS Labels report 🚀
